odra-casper/proxy-caller/bin/proxy_caller.rs

```diff
@@ -7,16 +7,20 @@ compile_error!("This binary only supports wasm32 target architecture!");
 
 extern crate alloc;
 
-use odra_casper_proxy_caller::ProxyCall;
-use odra_casper_wasm_env::casper_contract::contract_api::runtime::call_versioned_contract;
+use odra_casper_proxy_caller::{ensure_cargo_purse_is_empty, ProxyCall};
+
+use odra_casper_wasm_env::casper_contract::contract_api::runtime;
 
 #[no_mangle]
 fn call() {
     let proxy_call = ProxyCall::load_from_args();
-    let _: () = call_versioned_contract(
+    let _: () = runtime::call_versioned_contract(
         proxy_call.contract_package_hash,
         None,
         proxy_call.entry_point_name.as_str(),
         proxy_call.runtime_args
     );
+    if proxy_call.attached_value > 0.into() {
```

Comment:
You have this code duplicated. It could live under `proxy_call.post_call_pruse_check()`.